<?xml version="1.0"?>
<launch>
  <!-- arguments -->
  <arg name="nodelet_manager" default="velodyne_nodelet_manager" />
  <arg name="points_topic" default="/os1_cloud_node2/points" />
  <arg name="imu_topic" default="/os1_cloud_node2/imu" />

  <include file="$(find copilot_localization)/launch/kalman_filter/replay-localization-ukf.launch">
  </include>

  <param name="/use_sim_time" value="true"/>

  <!-- in case you use velodyne_driver, comment out the following line -->
  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)" args="manager" output="screen"/>

    <!-- globalmap_server_nodelet -->
    <node pkg="nodelet" type="nodelet" name="globalmap_server_nodelet" args="load hdl_localization/GlobalmapServerNodelet $(arg nodelet_manager)">
        <!-- <param name="globalmap_pcd" value="$(find hdl_localization)/data/map.pcd" /> -->
        <param name="globalmap_pcd" value="/home/simulator/highway_map6.pcd" />
      <param name="downsample_resolution" value="0.4" />
    </node>

    <!-- hdl_localization_nodelet -->
    <node pkg="nodelet" type="nodelet" name="hdl_localization_nodelet" args="load hdl_localization/HdlLocalizationNodelet $(arg nodelet_manager)">
        <remap from="/velodyne_points" to="$(arg points_topic)"/>
        <remap from="/gpsimu_driver/imu_data" to="$(arg imu_topic)"/>
        <!-- imu settings -->
        <!-- during "cool_time", imu inputs are ignored -->
        <param name="use_imu" value="true" />
        <param name="invert_imu" value="false" />
        <param name="cool_time_duration" value="0.5" />
        <!-- ndt settings -->
        <!-- if NDT is slow for your PC, try DIRECT1 serach method, which is a bit unstable but extremely fast -->
        <param name="ndt_neighbor_search_method" value="DIRECT7" />
        <param name="ndt_resolution" value="2.0" />
        <param name="downsample_resolution" value="0.2" />
        <!-- if "specify_init_pose" is true, pose estimator will be initialized with the following params -->
        <!-- otherwise, you need to input an initial pose with "2D Pose Estimate" on rviz" -->
        <param name="specify_init_pose" value="true" />
        <!-- <param name="init_pos_x" value="-100.0" />
        <param name="init_pos_y" value="70.0" />
        <param name="init_pos_z" value="-9.0" />
        <param name="init_ori_w" value="0.489386226655" />
        <param name="init_ori_x" value="0.0" />
        <param name="init_ori_y" value="0.0" />
        <param name="init_ori_z" value="-0.872067154043" /> -->
        <param name="init_pos_x" value="-106.570678711" />
        <param name="init_pos_y" value="54.9457015991" />
        <param name="init_pos_z" value="-9.0" />
        <param name="init_ori_w" value="0.609241803684" />
        <param name="init_ori_x" value="0.0" />
        <param name="init_ori_y" value="0.0" />
        <param name="init_ori_z" value="-0.792984504668" />
    </node>

    <node name="rviz" pkg="rviz" type="rviz" required="true"
              args="-d $(find hdl_localization)/rviz/hdl_localization.rviz" />

</launch>
